addons/mail/models/mail_thread.py

```diff
@@ -623,8 +623,8 @@ def _message_track(self, fields_iter, initial_values_dict):
 
             # find subtypes and post messages or log if no subtype found
             subtype = record._track_subtype(
-                dict((col_name, initial_values_dict[record.id][col_name])
-                     for col_name in changes)
+                {col_name: initial_values_dict[record.id][col_name]
+                     for col_name in changes}
             )
```

Comment:
We can avoid the weird hanging, and the extra line (`_track_subtype` has no other parameter)
```suggestion
                {col_name: initial_values_dict[record.id][col_name] for col_name in changes})
```